### PR TITLE
 switch mpileup to bcftools and subset to non-ref

### PR DESCRIPTION
**Pull request by nickhsmith**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Can you remove the `.swp` file?

File: subworkflows/local/bam_variant_calling_mpileup/main.nf [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
         fasta                   // channel: [mandatory]
+        dict
 
     main:
 
     ch_versions = Channel.empty()
 
-    SAMTOOLS_MPILEUP(cram, fasta)
-    mpileup = SAMTOOLS_MPILEUP.out.mpileup.branch{
+    BCFTOOLS_MPILEUP(cram, fasta,true)
```
Comment (maxulysse) [inline subworkflows/local/bam_variant_calling_mpileup/main.nf:16]: Can you comment what is this `true` value for?
like we do for this subworkflow:
https://github.com/nf-core/sarek/blob/d854ccc64ca6e14366bddd5a771d7b25b96f61c3/workflows/sarek.nf#L494-L495

File: subworkflows/local/bam_variant_calling_mpileup/main.nf [hunk 2]
```diff
@@ -32,18 +39,33 @@ workflow BAM_VARIANT_CALLING_MPILEUP {
                                             num_intervals:  meta.num_intervals,
                                             patient:        meta.patient,
                                             sample:         meta.sample,
-                                            status:         meta.status,
                                             sex:            meta.sex,
+                                            status:         meta.status,
```
Comment (maxulysse) [inline subworkflows/local/bam_variant_calling_mpileup/main.nf:43]:
```suggestion
                                            status:         meta.status
```

**Comment by maxulysse** [comment]
Looks good, but I still have the same question from the previous PR:
- Should we remove the SAMTOOLS_PILEUP module then?
- Will it impact in anyway the CONTROLFREEC part?

**Comment by nickhsmith** [comment]
> Looks good, but I still have the same question from the previous PR:
> 
>     * Should we remove the SAMTOOLS_PILEUP module then?
> 
>     * Will it impact in anyway the CONTROLFREEC part?

1. I think it depends on 2
2. It seems like the controlfreec parts are failing now, so it does. Maybe we just use the bcftools version for germline variant calling
3.

**Comment by maxulysse** [comment]
OK, then let's figure out if controlfreec is really impacted or not first, and then deal accordingly

**Comment by nickhsmith** [comment]
> OK, then let's figure out if controlfreec is really impacted or not first, and then deal accordingly

It was a different format and as a result I have switched the **BCFTOOLS** mpileup command to only produce vcfs and when `--tools mpileup`. As a result controlfreec uses the original SAMTOOLS and only calls variants using BCFTOOLS if mpileup is designated specifically

**Comment by nickhsmith** [comment]
All green :)

**Review by maxulysse — APPROVED** [review]

**Pull request merged**